daemon/api_test.go

```diff
@@ -1705,7 +1705,9 @@ func (s *apiSuite) TestRefresh(c *check.C) {
 	snapstateUpdate = func(s *state.State, name, channel string, userID int, flags snapstate.Flags) (*state.TaskSet, error) {
 		calledFlags = flags
 		calledUserID = userID
-		installQueue = append(installQueue, name)
+		if flags.DevModeAllowed() {
```

Comment:
mostly because of the amount of duplicated setup boilerplate, but yeah, it'd probably be better split